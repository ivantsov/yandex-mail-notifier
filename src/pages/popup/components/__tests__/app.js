import React from 'react';
import renderer from 'react-test-renderer';
import {AppComponent} from '../app';

jest.mock('shared/utils/i18n', () => {
    return {
        text(key) {
            return key;
        },
        date(str) {
            return str;
        },
    };
});
jest.mock('../header', () => () => <div>Header</div>);
jest.mock('../list', () => () => <div>List</div>);

const baseProps = {
    user: {
        authorized: true,
        email: 'test@ya.ru',
    },
    messages: {
        unreadCount: 5,
        items: [
            {id: 1},
            {id: 2},
        ],
        loading: false,
        error: false,
    },
    loadMessages: jest.fn(),
    updateMessage: jest.fn(),
    openLink: jest.fn(),
    openSettings: jest.fn(),
    reloadApp: jest.fn(),
    openDonationLink: jest.fn(),
};

function render(props) {
    const component = renderer.create(
        <AppComponent
            {...baseProps}
            {...props}
        />,
    );

    return component.toJSON();
}

describe('popup/App', () => {
    it('authorized', () => {
        const tree = render();

        expect(tree).toMatchSnapshot();
        expect(baseProps.loadMessages).toHaveBeenCalledTimes(1);
    });

    it('not authorized', () => {
        const tree = render({
            user: {
                ...baseProps.user,
                authorized: false,
            },
        });

        expect(tree).toMatchSnapshot();
        expect(baseProps.loadMessages).not.toBeCalled();
    });
});
