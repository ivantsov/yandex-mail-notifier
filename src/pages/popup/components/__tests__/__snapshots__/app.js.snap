// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`popup/App authorized 1`] = `
<div>
  <div>
    Header
  </div>
  <div>
    List
  </div>
</div>
`;

exports[`popup/App not authorized 1`] = `
<div
  className="container"
>
  <p>
    popup.notAuthorized.title
  </p>
  <p>
    popup.notAuthorized.subTitle
  </p>
  <button
    className="btn"
    onClick={[Function]}
  >
    popup.notAuthorized.btnText
  </button>
</div>
`;
